src/components/date-range/date-range.test.js

```diff
@@ -0,0 +1,370 @@
+          }}
+        />
+      );
+
+      getDataElementByValue("input")
+        .eq(START_DATE_RANGE_INDEX)
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should check the endDateProps prop", () => {
+      CypressMountWithProviders(
+        <DateRangeCustom
+          endDateProps={{
+            disabled: true,
+          }}
+        />
+      );
+
+      getDataElementByValue("input")
+        .eq(END_DATE_RANGE_INDEX)
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it.each([["top"], ["bottom"], ["left"], ["right"]])(
+      "should check the tooltipPosition is set to %s",
+      (position) => {
+        CypressMountWithProviders(
+          <DateRangeCustom
+            m={9}
+            tooltipPosition={position}
+            startError={testCypressText}
+          />
+        );
+        icon().realHover();
+        getDataElementByValue("tooltip").should("have.css", position);
+      }
+    );
+  });
+
+  describe("check events for DateRange component", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it.each([[0], [1]])(
+      "should call onChange callback when a clear event is triggered on %s DateRange input",
+      (inputIndex) => {
+        CypressMountWithProviders(<DateRangeCustom onChange={callback} />);
+
+        getDataElementByValue("input")
+          .eq(inputIndex)
+          .clear()
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      }
+    );
+
+    it.each([[0], [1]])(
+      "should call onChange callback when a type event is triggered on %s DateRange input",
+      (inputIndex) => {
+        CypressMountWithProviders(<DateRangeCustom onChange={callback} />);
+
+        getDataElementByValue("input")
+          .eq(inputIndex)
+          .type("01/10/2022")
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(
+              callback.getCalls()[9].args[0].target.value[inputIndex].rawValue
+            ).to.equal("2022-10-01");
+            expect(
+              callback.getCalls()[9].args[0].target.value[inputIndex]
+                .formattedValue
+            ).to.equal("01/10/2022");
+          });
+      }
+    );
+
+    it("should call onBlur callback when a blur event is triggered", () => {
+      CypressMountWithProviders(<DateRangeCustom onBlur={callback} />);
+
+      getDataElementByValue("input")
+        .eq(0)
+        .tab()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.not.have.been.called;
+        })
+        .tab()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should check name and id props", () => {
```

Comment:
Should we check the value prop here as well?

Reply:
no, we have checked value in the previous tests